### ISSUE TITLE
no-invalid-template-strings should ignore backslash-prefixed template expressions

### ISSUE THREAD

**Issue by RevanProdigalKnight** [palantir/tslint#2738]
### Bug Report

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: grunt-tslint

#### TypeScript code being linted

```ts
new (Tab.mixins.Controlled('MemberComments',Tab.mixins.Templated('<div mbr-comment-tab=\${which}></div>',Tab.mixins.AlwaysHasData())))
```

with `tslint.json` configuration:

```json
{
	"extends": "tslint:recommended",
	"rules": {
		// TSLint Options
		// Typescript-specific Rules
		"adjacent-overload-signatures": true,
		"no-internal-module": true,
		"no-magic-numbers": [true, -1, 0, 1, 2],
		"no-namespace": [true, "allow-declarations"],
		"only-arrow-functions": [true, "allow-declarations", "allow-named-functions"],
		"typedef-whitespace": [true, { "call-signature": "nospace", "index-signature": "nospace", "parameter": "nospace", "property-declaration": "nospace", "variable-declaration": "nospace" },{ "call-signature": "onespace", "index-signature": "onespace", "parameter": "onespace", "property-declaration": "onespace", "variable-declaration": "onespace" }],
		"unified-signatures": true,
		// Functionality Rules
		"curly": true,
		"forin": true,
		"label-position": true,
		"no-arg": true,
		"no-bitwise": false,
		"no-conditional-assignment": false,
		"no-console": true,
		"no-construct": true,
		"no-debugger": true,
		"no-duplicate-super": true,
		"no-duplicate-variable": true,
		"no-empty": true,
		"no-eval": true,
		"no-for-in-array": true,
		"no-invalid-template-strings": true,
		"no-invalid-this": true,
		"no-misused-new": true,
		"no-string-throw": true,
		"no-switch-case-fall-through": true,
		"no-unsafe-finally": true,
		"no-unused-expression": [true, "allow-fast-null-checks"],
		"no-unused-variable": [true],
		"no-shadowed-variable": false,
		"no-use-before-declare": true,
		"no-var-keyword": true,
		"radix": true,
		"restrict-plus-operands": true,
		"strict-type-predicates": true,
		"triple-equals": true,
		"typeof-compare": true,
		"use-isnan": true,
		// Maintainability Rules
		"cyclomatic-complexity": 20,
		"eofline": true,
		"indent": [true, "tabs"],
		"linebreak-style": [true, "LF"],
		"no-require-imports": true,
		"no-trailing-whitespace": true,
		"object-literal-sort-keys": true,
		"prefer-const": [true, { "destructuring": "all" }],
		"trailing-comma": false,
		// Style Rules
		"arrow-parens": [true, "ban-single-arg-parens"],
		"array-type": [true, "array"],
		"arrow-return-shorthand": true,
		"callable-types": true,
		"class-name": true,
		"comment-format": [true, "check-space"],
		"import-spacing": true,
		"interface-name": true,
		"interface-over-type-literal": true,
		"newline-before-return": true,
		"new-parens": true,
		"no-angle-bracket-type-assertion": false,
		"no-unnecessary-callback-wrapper": true,
		"no-unnecessary-qualifier": true,
		"object-literal-key-quotes": [true, "as-needed"],
		"object-literal-shorthand": true,
		"one-line": [true, "check-catch", "check-finally", "check-else", "check-open-brace"],
		"one-variable-per-declaration": false,
		"prefer-method-signature": true,
		"quotemark": [true, "single", "avoid-escape"],
		"semicolon": true,
		"variable-name": [true, "check-format", "allow-leading-underscore", "allow-pascal-case", "ban-keywords"],
		"whitespace": false
	}
}
```

#### Actual behavior
`>> ERROR: src/main/ts/padss-main/tabs.ts[423, 90]: Interpolation will only work for template strings.`

#### Expected behavior
The rule should ignore the template replacement expression because the developer has already acknowledged that they realize this is not a template string by adding the preceding `\`.

Even if it were a template string, adding the preceding `\` causes the template expression to be ignored in the template string anyways.

**Comment by aervin** [comment]
Haven't been able to find an open PR for this one so I'll give it a try.
